### PR TITLE
Tacotron2, yolov3*: benchmark coverage for custom devices.

### PR DESCRIPTION
**Pull request by weishi-deng**
Works for Roadmap https://github.com/pytorch/benchmark/issues/1293 to increase benchmark coverage.

For these 5 models:tacotron2, yolov3, nvidia_deeprecommender, LearningToPoint and pytorch_CycleGAN_and_pix2pix, 
when running on custom devices except for CPU and CUDA(e.g. XPU), it will raise the error as it's hard-coded with CPU/CUDA backends.
In this PR, we accept the device args as a param within the training process and inference process which will cover the model initializing and data transposition for these custom devices.

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
@xuzhao9 , @atalman, This PR is a follow-up of Intel GPU-enabling as described in the RFC(https://github.com/pytorch/pytorch/issues/114723).

So far, we have landed Intel GPU runtime and Inductor and limited ATen operations. We are evaluating the dynamo benchmarks and have obtained a reasonable accuracy pass rate for HF. However, for TorchBench, we encountered some functionality issues with the Intel GPU as Intel GPU is a new device, and we need to enable it. 

This PR intends to enable Intel GPU and the fix the functionality issues.

**Review by chuanqi129 — APPROVED** [review]

**Comment by xuzhao9** [comment]
The CI failure is pre-existing and is not related to this PR.

**Review by xuzhao9 — APPROVED** [review]
